sycl/test-e2e/ESIMD/accessor_local.cpp

```diff
@@ -0,0 +1,134 @@
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip || esimd_emulator
+// UNSUPPORTED: gpu-intel-gen9 && windows
+//
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// TODO: Enable the test when GPU driver is ready/fixed.
+// XFAIL: gpu
+
+// This test verifies usage of local_accessor methods operator[]
+// and get_pointer().
+
+#include "esimd_test_utils.hpp"
+
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/sycl.hpp>
+
+#include <iostream>
+
+using namespace sycl;
+using namespace sycl::ext::intel::esimd;
+
+constexpr int VL = 16;
+
+template <typename T, bool TestSubscript>
+bool test(queue Q, uint32_t LocalRange, uint32_t GlobalRange) {
+  std::cout << "Running case: T=" << esimd_test::type_name<T>()
+            << ", TestSubscript=" << TestSubscript << std::endl;
+
+  // The test is going to use (LocalRange * VL) elements of T type.
+  auto Dev = Q.get_device();
+  auto DeviceSLMSize = Dev.get_info<sycl::info::device::local_mem_size>();
+  if (DeviceSLMSize < LocalRange * VL * sizeof(T)) {
+    // Report an error - the test needs a fix.
+    std::cerr << "Error: Test needs more SLM memory than device has!"
```

Comment:
I intentionally made it an error. The amount of SLM memory used by the test is pretty small now.
If the test eventually fails it means some unusual case was met, and it requires attention and tuning of the test rather than silent pass.